addons/hr_payroll/models/hr_payroll.py

```diff
@@ -2,315 +2,237 @@
+
+    @api.multi
+    def get_all_structures(self):
         """
-        @param contract_ids: list of contracts
         @return: the structures linked to the given contracts, ordered by hierachy (parent=False first, then first level children and so on) and without duplicata
         """
-        structure_ids = [contract.struct_id.id for contract in self.browse(cr, uid, contract_ids, context=context) if contract.struct_id]
-        if not structure_ids:
+        structures = self.mapped('struct_id')
+        if not structures:
             return []
-        return list(set(self.pool.get('hr.payroll.structure')._get_parent_structure(cr, uid, structure_ids, context=context)))
-
+        return list(set(structures._get_parent_structure().ids))
 
-class contrib_register(osv.osv):
-    '''
-    Contribution Register
-    '''
 
+class ContribRegister(models.Model):
     _name = 'hr.contribution.register'
     _description = 'Contribution Register'
 
-    _columns = {
-        'company_id':fields.many2one('res.company', 'Company'),
-        'partner_id':fields.many2one('res.partner', 'Partner'),
-        'name':fields.char('Name', required=True, readonly=False),
-        'register_line_ids':fields.one2many('hr.payslip.line', 'register_id', 'Register Line', readonly=True),
-        'note': fields.text('Description'),
-    }
-    _defaults = {
-        'company_id': lambda self, cr, uid, context: \
-                self.pool.get('res.users').browse(cr, uid, uid,
-                    context=context).company_id.id,
-    }
-
-
-class hr_salary_rule_category(osv.osv):
-    """
-    HR Salary Rule Category
-    """
+    company_id = fields.Many2one('res.company', string='Company',
+        default=lambda self: self.env['res.company']._company_default_get())
+    partner_id = fields.Many2one('res.partner', string='Partner')
+    name = fields.Char(required=True)
+    register_line_ids = fields.One2many('hr.payslip.line', 'register_id',
+        string='Register Line', readonly=True)
+    note = fields.Text(string='Description')
 
+
+class HrSalaryRuleCategory(models.Model):
     _name = 'hr.salary.rule.category'
     _description = 'Salary Rule Category'
-    _columns = {
-        'name':fields.char('Name', required=True, readonly=False),
-        'code':fields.char('Code', size=64, required=True, readonly=False),
-        'parent_id':fields.many2one('hr.salary.rule.category', 'Parent', help="Linking a salary category to its parent is used only for the reporting purpose."),
-        'children_ids': fields.one2many('hr.salary.rule.category', 'parent_id', 'Children'),
-        'note': fields.text('Description'),
-        'company_id':fields.many2one('res.company', 'Company', required=False),
-    }
-
-    _defaults = {
-        'company_id': lambda self, cr, uid, context: \
-                self.pool.get('res.users').browse(cr, uid, uid,
-                    context=context).company_id.id,
-    }
-
-
-class one2many_mod2(fields.one2many):
```

Comment:
I think it is basically a one2many with a domain. Before the domain attr was added on fields.